app/components/primer/octicon_component.rb

```diff
@@ -27,20 +29,85 @@ class OcticonComponent < Primer::Component
     # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
     def initialize(icon_name = nil, icon: nil, size: SIZE_DEFAULT, **system_arguments)
-      @icon = icon_name || icon
+      icon_key = icon_name || icon
+      cache_key = [icon_key, size, system_arguments.slice(:height, :width)].to_s
```

Comment:
```suggestion
      cache_key = [icon_key, size, system_arguments.slice(:height, :width)].join("_")
```

Minor, but I think that will create a cache key that doesn't look like an array.